service/tablet_allocator.cc

```diff
@@ -836,7 +837,13 @@ class load_balancer {
 
         std::vector<repair_plan> plans;
         auto migration_tablet_ids = co_await mplan.get_migration_tablet_ids();
+        const auto& tables_waiting_for_resize_finalization = mplan.resize_plan().finalize_resize;
         for (auto&& [table, tmap_] : _tm->tablets().all_tables()) {
+            // Skip tables waiting for resize finalization
+            if (tables_waiting_for_resize_finalization.contains(table)) {
+                continue;
```

Comment:
If we don't have per-table finalization, we should stop scheduling repair as soon as there is any table awaiting finalization, same as for migrations: `!plan.resize_plan().finalize_resize.empty()`

Those repairs will block finalization, but also will block load-balancing, which is inhibited by finalization request, which is inhibited by repairs.

Reply:
Ack. Done.